FrontEnd/src/HomePage/MainPage.jsx

```diff
@@ -1,11 +1,15 @@
 import { useEffect, useState } from "react";
 import { fetchUserFeed } from "../UserAuthentication/FirestoreDB";
 import CategoryList from "./CategoryList";
+import PostsPreview from "./PostsPreview";
+import PostFullDisplay from "../PostFullDisplay";
 import "./MainPage.css";
 
-function MainPage({ userUID }) {
+function MainPage({ userUID, userData }) {
   const [userFeed, setUserFeed] = useState([]);
   const [isLoading, setIsLoading] = useState(true);
+  const [isModalShown, setIsModalShown] = useState(false);
```

Comment:
For better code readability it would be useful to make the variable name more descriptive, ex: 
```suggestion
  const [isPostDetailsModalShown, setIsPostDetailsModalShown] = useState(false);
```

Otherwise it isn't clear at this point which modal this will open.